test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +178,298 @@ int suiteTearDown( int numFailures )
 {
     return numFailures;
 }
+
+/* ========================================================================== */
+
+/**
+ * @brief Test that MQTTAgentCommand_ProcessLoop() works as intended.
+ */
+void test_MQTTAgentCommand_ProcessLoop( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    mqttStatus = MQTTAgentCommand_ProcessLoop( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Publish() success case with QoS0.
+ */
+void test_MQTTAgentCommand_Publish_QoS0_success( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTPublishInfo_t publishInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    /* Initializing QOS. */
+    publishInfo.qos = MQTTQoS0;
+
+    MQTT_Publish_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Publish( &mqttAgentContext, &publishInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+    TEST_ASSERT_FALSE( returnFlags.addAcknowledgment );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Publish() success case with QoS1.
+ */
+void test_MQTTAgentCommand_Publish_QoS1_success( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTPublishInfo_t publishInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    /* Initializing QOS. */
+    publishInfo.qos = MQTTQoS1;
+
+    MQTT_GetPacketId_IgnoreAndReturn( 1 );
+    MQTT_Publish_ExpectAndReturn( &mqttAgentContext, &publishInfo, 1, MQTTSuccess );
```

Comment:
Different pointer types
```suggestion
    MQTT_Publish_ExpectAndReturn( &( mqttAgentContext.mqttContext ), &publishInfo, 1, MQTTSuccess );
```